pddl/_validation.py

```diff
@@ -234,6 +252,86 @@ def _(self, predicate: Predicate) -> None:
+        """Check types annotations of a PDDL numeric function."""
+        self.check_type(function.terms)
+
+    @check_type.register
+    def _(self, _: NumericValue) -> None:
+        """Check types annotations of a PDDL numeric value operator."""
+        return None
+
+    @check_type.register
+    def _(self, equal_to: FunctionEqualTo) -> None:
+        """Check types annotations of a PDDL numeric equal to operator."""
+        self.check_type(equal_to.operands)
+
+    @check_type.register
+    def _(self, lesser: LesserThan) -> None:
+        """Check types annotations of a PDDL numeric lesser operator."""
+        self.check_type(lesser.operands)
+
+    @check_type.register
+    def _(self, lesser_equal: LesserEqualThan) -> None:
+        """Check types annotations of a PDDL numeric lesser or equal operator."""
+        self.check_type(lesser_equal.operands)
+
+    @check_type.register
+    def _(self, greater: GreaterThan) -> None:
+        """Check types annotations of a PDDL numeric greater operator."""
+        self.check_type(greater.operands)
+
+    @check_type.register
+    def _(self, greater_equal: GreaterEqualThan) -> None:
+        """Check types annotations of a PDDL numeric greater or equal to."""
+        self.check_type(greater_equal.operands)
+
+    @check_type.register
+    def _(self, assign: Assign) -> None:
+        """Check types annotations of a PDDL numeric assign to."""
+        self.check_type(assign.operands)
+
+    @check_type.register
+    def _(self, scale_up: ScaleUp) -> None:
+        """Check types annotations of a PDDL numeric assign to."""
+        self.check_type(scale_up.operands)
+
+    @check_type.register
+    def _(self, scale_down: ScaleDown) -> None:
+        """Check types annotations of a PDDL numeric assign to."""
+        self.check_type(scale_down.operands)
+
+    @check_type.register
+    def _(self, increase: Increase) -> None:
+        """Check types annotations of a PDDL numeric increase to."""
+        self.check_type(increase.operands)
+
+    @check_type.register
+    def _(self, decrease: Decrease) -> None:
+        """Check types annotations of a PDDL numeric decrease operator."""
+        self.check_type(decrease.operands)
+
+    @check_type.register
+    def _(self, minus: Minus) -> None:
+        """Check types annotations of a PDDL numeric minus operator."""
+        self.check_type(minus.operands)
+
+    @check_type.register
+    def _(self, plus: Plus) -> None:
+        """Check types annotations of a PDDL numeric plus operator."""
+        self.check_type(plus.operands)
+
+    @check_type.register
+    def _(self, times: Times) -> None:
+        """Check types annotations of a PDDL numeric times operator."""
+        self.check_type(times.operands)
+
+    @check_type.register
+    def _(self, divide: Divide) -> None:
+        """Check types annotations of a PDDL numeric divide operator."""
+        self.check_type(divide.operands)
```

Comment:
note, you could have used just `def _(self, binary_fun: BinaryFunction): ...`, the dispatch would have matched all subclasses of `BinaryFunction`

Reply:
You're right, thank you! I just forgot about the `BinaryFunction` and wrote all blindly 😅

Reply:
It's fine as you wrote it. In the future, we might add further validation for specific operators; hence, we might need (some of) those specific functions. Do what you wish :)

Reply:
Yep, for now, I think it's better to wrap them all under the binary function so to have less code.

Reply:
This resolved?